### PR TITLE
[SYNCOPE-160] Importing preliminary work

### PR DESCRIPTION
**Pull request by ilgrosso**

### CODE REVIEW

File: core/provisioning-java/src/main/java/org/apache/syncope/core/provisioning/java/data/PolicyDataBinderImpl.java [hunk 1]
```diff
@@ -192,6 +199,54 @@
...
+            if (result == null) {
+                result = (T) entityFactory.newEntity(AuthPolicy.class);
+            }
+
+            AuthPolicy authPolicy = AuthPolicy.class.cast(result);
+            AuthPolicyTO authPolicyTO = AuthPolicyTO.class.cast(policyTO);
+
+            authPolicy.setName(authPolicyTO.getKey());
+            Implementation configurtion = implementationDAO.find(authPolicyTO.getConfiguration());
+            if (configurtion == null) {
```
Comment (mat-ale) [inline core/provisioning-java/src/main/java/org/apache/syncope/core/provisioning/java/data/PolicyDataBinderImpl.java:212]: just a small naming typo (-> `configurtion`)

File: core/provisioning-java/src/main/java/org/apache/syncope/core/provisioning/java/data/PolicyDataBinderImpl.java [hunk 2]
```diff
@@ -192,6 +199,54 @@
...
+        } else if (policyTO instanceof AccessPolicyTO) {
+            if (result == null) {
+                result = (T) entityFactory.newEntity(AccessPolicy.class);
+            }
+
+            AccessPolicy accessPolicy = AccessPolicy.class.cast(result);
+            AccessPolicyTO accessPolicyTO = AccessPolicyTO.class.cast(policyTO);
+
+            accessPolicy.setName(accessPolicyTO.getKey());
+            Implementation configurtion = implementationDAO.find(accessPolicyTO.getConfiguration());
```
Comment (mat-ale) [inline core/provisioning-java/src/main/java/org/apache/syncope/core/provisioning/java/data/PolicyDataBinderImpl.java:227]: same typo here

File: core/provisioning-java/src/main/java/org/apache/syncope/core/provisioning/java/data/PolicyDataBinderImpl.java [hunk 3]
```diff
@@ -192,6 +199,54 @@
...
+        } else if (policyTO instanceof AttrReleasePolicyTO) {
+            if (result == null) {
+                result = (T) entityFactory.newEntity(AttrReleasePolicy.class);
+            }
+
+            AttrReleasePolicy attrReleasePolicy = AttrReleasePolicy.class.cast(result);
+            AttrReleasePolicyTO attrReleasePolicyTO = AttrReleasePolicyTO.class.cast(policyTO);
+
+            attrReleasePolicy.setName(attrReleasePolicyTO.getKey());
+            Implementation configurtion = implementationDAO.find(attrReleasePolicyTO.getConfiguration());
```
Comment (mat-ale) [inline core/provisioning-java/src/main/java/org/apache/syncope/core/provisioning/java/data/PolicyDataBinderImpl.java:243]: same typo here

File: core/persistence-api/src/main/java/org/apache/syncope/core/persistence/api/entity/auth/AuthModuleItem.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.syncope.core.persistence.api.entity.auth;
+
+import org.apache.syncope.core.persistence.api.entity.resource.Item;
+
+public interface AuthModuleItem extends Item {
+
+}
```
Comment (mat-ale) [inline core/persistence-api/src/main/java/org/apache/syncope/core/persistence/api/entity/auth/AuthModuleItem.java:25]: Not completely sure about this, but for the moment it should work! @ilgrosso what do you think?

Comment (ilgrosso) [reply core/persistence-api/src/main/java/org/apache/syncope/core/persistence/api/entity/auth/AuthModuleItem.java:25]: if you took [OIDCProviderItem](https://github.com/apache/syncope/blob/master/ext/oidcclient/persistence-api/src/main/java/org/apache/syncope/core/persistence/api/entity/OIDCProviderItem.java) or [SAML2IdPItem](https://github.com/apache/syncope/blob/master/ext/saml2sp/persistence-api/src/main/java/org/apache/syncope/core/persistence/api/entity/SAML2IdPItem.java) as model, then there is something obviously missing, e.g. the owning entity; I would have expected:

```java
public interface AuthModuleItem extends Item {

    AuthModule getAuthModule();

    void setAuthModule(AuthModule authModule);
}
```

Comment (ilgrosso) [reply core/persistence-api/src/main/java/org/apache/syncope/core/persistence/api/entity/auth/AuthModuleItem.java]: Fixed with last commit.

**Pull request merged**